packages/core/useWorkerWebFunction/lib/jobRunner.ts

```diff
@@ -0,0 +1,29 @@
+
+/**
+ * This function accepts as a parameter a function "userFunc"
+ * And as a result returns an anonymous function.
+ * This anonymous function, accepts as arguments,
+ * the parameters to pass to the function "useArgs" and returns a Promise
+ * This function can be used as a wrapper, only inside a Worker
+ * because it depends by "postMessage".
+ *
+ * @param {Function} userFunc {Function} fn the function to run with web worker
+ *
+ * @returns {Function} returns a function that accepts the parameters
+ * to be passed to the "userFunc" function
+ */
+const jobRunner = (userFunc: Function) => (e: MessageEvent) => {
+  const userFuncArgs = e.data[0]
+
+  return Promise.resolve(userFunc(...userFuncArgs))
```

Comment:
```suggestion
  // eslint-disable-next-line prefer-spread
  return Promise.resolve(userFunc.apply(undefined, userFuncArgs))
```

I think this can solve the `_toConsumableArray is not defined` problem. I think this may vary in different babel settings, but for this, I think it's not necessary to use spread since it may have a higher chance to introduce errors.

Reply:
Ya that fixed it. But this means, you need to write es5 code to be secure the code works fine at least in this setup.

This limits the ability of the hook. Maybe you also dont wanna add it, because it can lead to frustrating bugs, if you dont read the docu enough

Reply:
I think this may depend on the userland babel/ts config. There are tons of limitations in web worker such as dom access as well, I think adding a notice in the docs would be fine for this. (and maybe jsdoc as well, so people can see the note in there editor)